### PR TITLE
Implement TenantTemplate

### PR DESCRIPTION
**Pull request by corvus-ch**
This is the implementation for #136.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: pkg/pipeline/tenant_steps.go [hunk 1]
```diff
@@ -83,3 +84,25 @@ func setGlobalGitRepoURL(obj PipelineObject, data *ExecutionContext) ExecutionRe
...
+
+func applyTemplateFromTenantTemplate(obj PipelineObject, data *ExecutionContext) ExecutionResult {
+	key := types.NamespacedName{Name: "default", Namespace: obj.GetObjectMeta().GetNamespace()}
+	template := &synv1alpha1.TenantTemplate{}
+	if err := data.Client.Get(context.TODO(), key, template); err != nil {
+		// The absence of a template is not an error.
+		// It simply means that there is nothing to do.
+		// TODO Consider to log this case. Where to get a logger from?
+		return ExecutionResult{}
+	}
```
Comment (ccremer) [inline pkg/pipeline/tenant_steps.go:96]: We can at least ignore the IsNotFound error. Other errors we can hardly ignore like temporary, e.g. network-related errors that may in a later reconciliation be resolved again.
We might consider Conditions here to report back errors, also as a means for the lieutenant-api to respect failed reconciliations (e.g. if there's no Ready condition, throw an API error etc)

File: pkg/pipeline/tenant_steps_test.go [hunk 2]
```diff
@@ -171,3 +172,68 @@ func Test_updateTenantGitRepo(t *testing.T) {
...
+			Spec: synv1alpha1.TenantSpec{
+				DisplayName: "My Tenant",
+			},
+		}
+		tenantOut := &synv1alpha1.Tenant{
+			Spec: synv1alpha1.TenantSpec{
+				DisplayName: "My Tenant",
+				DeletionPolicy: synv1alpha1.DeletePolicy,
+			},
+		}
```
Comment (ccremer) [inline pkg/pipeline/tenant_steps_test.go:231]: I'm missing a case where the annotation `lieutenant.syn.tools/tenant-template: "default"` gets added

File: docs/modules/ROOT/pages/how-tos/create-tenant.adoc [hunk 3]
```diff
@@ -27,6 +27,16 @@ spec:
...
+== Tenant Templating
+
+Values common to all new tenants, can be put into a template.
+When a new tenant gets created, the operator looks for a
+`TenantTemplate` object named `default`.
+A `TenantTemplate` has the same spec data structure as a `Tenant`.
+If present, the templates values from the template will be merged into the tenant.
+The values set on the tenant will take precedence.
+In addition, a reference to the template will be added as an annotation.
+The annotation is named `lieutenant.syn.tools/tenant-template`, and the value will be the name of the tenant.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-tos/create-tenant.adoc:39]:
```suggestion
== Tenant Templating

Values common to all new tenants, can be put into a `TenantTemplate`.
When a new tenant gets created, the operator looks for a `TenantTemplate` object named `default` in the same namespace as `Tenant`.
A `TenantTemplate` has the same spec data structure as a `Tenant`.
If present, the template's values from the template will be merged into the `Tenant` spec.
The values set on the tenant will take precedence.
In addition, a reference to the template will be added as an annotation.
The annotation is named `lieutenant.syn.tools/tenant-template`, and the value will be the name of the `TenantTemplate`.
```

**Review by ccremer — APPROVED** [review]

**Review by Kidswiss — APPROVED** [review]
LGTM

**Pull request merged**